Fix feature name validation regex

theneva opened the pull request:
Those underscores broke when support for underscores was added in https://github.com/Unleash/unleash/pull/267. The `-` was no longer at the end of the expression, and thus changed meaning to `from \ to \`.

I found no tests for valid names; how do I best write a test for this?

SimenB commented on lib/routes/admin-api/feature.js:
```diff
@@ -115,8 +115,8 @@ module.exports.router = function(config) {
-            .checkBody('name', 'Name must match format ^[0-9a-zA-Z\\.\\-\\_]+$')
-            .matches(/^[0-9a-zA-Z\\.\\-\\_]+$/i);
+            .checkBody('name', 'Name must match format ^[0-9a-zA-Z\.\-\_]+$')
+            .matches(/^[0-9a-zA-Z\.\-\_]+$/i);
```
`/i` makes no sense

SimenB replied:
Also, just extract it into a constant so we don't have to duplicate it (you can run `.source` on a regex to get the underlying string out)

theneva replied:
I agree, will fix. The regex can be simplified to `/^[0-9a-zA-Z\-._]+$/`

ivarconr commented:
For some reason we have not implemented the same check when creating strategies, which I find stupid. 

Would you mind adding a similar check inside https://github.com/Unleash/unleash/blob/master/lib/routes/admin-api/strategy.js also? Then it also would be nice to reuse this regex in both routes.

theneva commented:
I've written a test for some commons name patterns I could think of, simplified the regex a bit, and extracted it to a constant. I think that's enough for this PR—it fixes the bug, and should be easy enough to build the rest of the stuff discussed in this PR upon.

I realise that I could have avoided the need to escape the `-` by tacking it on to the end. However, that's what introduced the bug in the first place, so I'll go with the better-safe-than-sorry strategy on this one.

How does this look now? cc @SimenB @ivarconr
